src/mkdocs_gallery/scrapers.py

```diff
@@ -192,32 +192,32 @@ def matplotlib_scraper(block, script: GalleryScript, **kwargs):
 
         # save the figures, and populate the srcsetpaths
         try:
-            fig.savefig(image_path, **these_kwargs)
+            fig.savefig(str(image_path), **these_kwargs)
             dpi0 = matplotlib.rcParams["savefig.dpi"]
             if dpi0 == "figure":
                 dpi0 = fig.dpi
             dpi0 = these_kwargs.get("dpi", dpi0)
-            srcsetpaths = {0: image_path}
+            srcsetpaths = {0.0: image_path}
```

Comment:
If using a float here is not related to the fix for #75, I would prefer to keep this code section identical to the sphinx-gallery original code, for maintenability reasons (it will be easier to cherry-pick some changes)

https://github.com/sphinx-gallery/sphinx-gallery/blob/f5cc4613b17e22ed593b8f30fcd5f879b33d53d5/sphinx_gallery/scrapers.py#L182-L209

Reply:
Have reverted all unneed changed, didnt realise the were to keep compatibility with sphinx gallery. Also, damn, never realised sphinx-gallery was such a mess